server/src/lib/mutex.h

```diff
@@ -0,0 +1,52 @@
+#pragma once
+#include <fty/expected.h>
+#include <optional>
+#include <atomic>
+#include <mutex>
+#include <condition_variable>
+
+namespace fty::storage {
+
+class Mutex
+{
+public:
+    Mutex();
+    ~Mutex();
+
+    enum class AccessType{
+        READ,
+        WRITE
+    };
+
+    class WRITE {
+        Mutex * m;
```

Comment:
Better to initialize in definition.
Mutex* m = nullptr;
bool locked = false; 

And as it private members, m_ prefix.